contracts/MechMarketplace.sol

```diff
@@ -365,6 +371,29 @@ contract MechMarketplace is IErrorsMarketplace {
         emit SetMechFactoryStatuses(mechFactories, statuses);
     }
 
+    // Check and escrow delivery rate
+    function _checkAndEscrowDeliveryRate(address mech) internal {
+        IMech.MechType mechType = IMech(mech).mechType();
+        uint256 maxDeliveryRate = IMech(mech).maxDeliveryRate();
+
+        // Check delivery rates
+        if (mechType == IMech.MechType.FixedPrice) {
+            // Check the request delivery rate for a fixed price
+            if (msg.value < maxDeliveryRate) {
+                revert InsufficientBalance(msg.value, maxDeliveryRate);
+            }
+        } else {
+            // Check that there is no incoming deposit
+            if (msg.value > 0) {
+                revert NoDepositAllowed(msg.value);
+            }
+
+            // TODO Add onERC1155Received to the receiving contract
+            // Get max subscription rate for escrow from sender
+            IERC1155(subscriptionNFT).safeTransferFrom(msg.sender, address(this), subscriptionTokenId, maxDeliveryRate, "");
```

Comment:
> Why does this happen here? should happen on delivery

Seems that this is transferring to the marketplace and then on delivery it is transfered to the balance of the mech who delivers. But true that it probably payment could probably be transfered directly from the requester to the mech's balance without having the marketplace as an intermediate holder.

Reply:
We don't need to transfer anything, just check, because the subscription is already paid, although I'd prefer to transfer credits to MM because by the time the payment needs to be done in bulk - there might not be credits left, and this is the issue.

Reply:
Let the micropayments and microcalculations be managed by MM processors, and the fee is applied when the mech pulls with a single transfer to that mech.

Reply:
This is about any external tokens - unless taken into hold - there's no guarantee the payment is going to be there by the checkout time.